flow/connectors/utils/array.go

```diff
@@ -29,3 +29,19 @@ func ArrayChunks[T any](slice []T, size int) [][]T {
 
 	return partitions
 }
+
+func ArraysHaveOverlap[T comparable](first, second []T) bool {
+	lookup := make(map[T]struct{})
+
+	for _, element := range first {
```

Comment:
I generally expect the first parameter will be larger than the second (see, for example, how callers have first parameter as existing tables, & second parameter as new tables). Better off constructing set that's the size of second slice

Reply:
swapped around